tests/test_theseus_report.py

```
@@ -0,0 +1,171 @@
+            uptime_path = Path(tmp) / "uptime"
+            uptime_path.write_text("nan 123.4\n", encoding="utf-8")
+            report_data = report.build_report(FixtureRunner(), FIXTURES / "os-release", uptime_path=uptime_path)
+        states = {item["collector"]: item["state"] for item in report_data["collection_status"]}
+        self.assertEqual(states["uptime"], "failed")
+        self.assertNotIn("uptime_seconds", report_data["volatile"])
+        json.dumps(report_data, allow_nan=False)
+
+    def test_collect_removes_temporary_file_and_output_directory_on_write_failure(self):
+        real_named_temp_file = tempfile.NamedTemporaryFile
+
+        def broken_named_temp_file(*args, **kwargs):
```

Ruff (0.16.2)

[warning] Missing return type annotation for private function `broken_named_temp_file`

(ANN202)